xwiki-platform-core/xwiki-platform-realtime/xwiki-platform-realtime-wiki/xwiki-platform-realtime-wiki-test/xwiki-platform-realtime-wiki-test-docker/src/test/it/org/xwiki/realtime/wiki/test/ui/RealtimeWikiEditorIT.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.realtime.wiki.test.ui;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.WindowType;
+import org.xwiki.realtime.wiki.test.po.RealtimeWikiEditor;
+import org.xwiki.test.docker.junit5.TestReference;
+import org.xwiki.test.docker.junit5.UITest;
+import org.xwiki.test.ui.TestUtils;
+
+/**
+ * Functional tests for the real-time Wiki editor.
+ * 
+ * @version $Id$
+ * @since 15.5.4
+ * @since 15.10
+ */
+@UITest(
+    extraJARs = {
+        // The WebSocket end-point implementation based on XWiki components needs to be installed as core extension.
+        "org.xwiki.platform:xwiki-platform-websocket",
+    }
+)
+class RealtimeWikiEditorIT
+{
+    protected static String firstTabHandle;
+    
+    @BeforeAll
+    void beforeAll(TestUtils setup) {
+        firstTabHandle = setup.getDriver().getWindowHandle();
+        
+        setup.loginAsSuperAdmin();
+        
+        // We need to set the realtime wiki editor as the default wiki editor.
+        setup.addClassProperty("XWiki", "EditorBindingClass", "dataType", "String");
+        setup.addClassProperty("XWiki", "EditorBindingClass", "roleHint", "String");
```

Comment:
Shouldn't this class be present? Couldn't we add the module that provides it as a test dependency?

Reply:
The component is provided by `xwiki-platform-edit-default`, which is a dependency of `xwiki-platform-realtime-wiki-ui`. I'm not sure why it doesn't work without it, I'll investigate a bit more.

Reply:
It was indeed not needed, thanks.